src/shared/particles/base_particles.cpp

```diff
@@ -414,17 +414,17 @@ void BaseParticles::readFromXmlForReloadParticle(std::string &filefullpath)
 template <typename DataType>
 struct copyVariablesFromDeviceOperation
 {
-    void operator()(DeviceVariables &device_variables,
-                    ParticleData &particle_data,
-                    ParticleVariables &particle_variables,
+    void operator()(const DeviceVariables &device_variables,
+                    const ParticleData &particle_data,
+                    const ParticleVariables &particle_variables,
                     execution::ExecutionEvent &event) const
     {
         constexpr int type_index = DataTypeIndex<DataType>::value;
         for (DiscreteVariable<DataType> *variable : std::get<type_index>(particle_variables))
         {
             if constexpr (DataTypeEquivalence<DataType>::type_defined)
             {
-                DeviceVariable<typename DataTypeEquivalence<DataType>::device_type> *device_variable = findVariableByName<typename DataTypeEquivalence<DataType>::device_type>(device_variables, variable->Name());
+                auto *device_variable = findVariableByName<typename DataTypeEquivalence<DataType>::device_type>(device_variables, variable->Name());
```

Comment:
When you use "auto", please take into consideration that whether it will decrease the readability, while may not improve the feasibility.